src/SDKs/Azure.Core.Identity/data-plane/Azure.Core.Identity/IdentityClient.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.Pipeline.Policies;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Identity
+{
+    public class IdentityClientOptions : HttpClientOptions
+    {
+        private const string DefaultAuthorityHost = "https://login.microsoftonline.com/";
+
+        public RetryPolicy RetryPolicy { get; set; }
+
+        public HttpPipelinePolicy LoggingPolicy { get; set; }
+
+        public string AuthorityHost { get; set; }
+
+        public IdentityClientOptions()
+        {
+            AuthorityHost = DefaultAuthorityHost;
+            LoggingPolicy = Core.Pipeline.Policies.LoggingPolicy.Shared;
+            RetryPolicy = new ExponentialRetryPolicy()
+            {
+                Delay = TimeSpan.FromMilliseconds(800),
+                MaxRetries = 3
+            };
+        }
+    }
+
+    internal class IdentityClient
+    {
+        private readonly IdentityClientOptions _options;
+        private readonly HttpPipeline _pipeline;
+        
+
+        public IdentityClient(IdentityClientOptions options = null)
+        {
+            _options = options ?? new IdentityClientOptions();
+
+            _pipeline = HttpPipeline.Build(options,
+                    options.ResponseClassifier,
+                    options.RetryPolicy,
+                    ClientRequestIdPolicy.Singleton,
+                    options.LoggingPolicy,
+                    BufferResponsePolicy.Singleton);
+        }
+
+        public async Task<AuthenticationResponse> AuthenticateAsync(string tenantId, string clientId, string clientSecret, IEnumerable<string> scopes, CancellationToken cancellationToken = default)
+        {
+            Uri authUri = new Uri(_options.AuthorityHost + tenantId + "/oauth2/token");
+
+            using (var request = _pipeline.CreateRequest())
+            {
+                var body = new Dictionary<string, string>() {
+                    { "response_type", "token" },
+                    { "grant_type", "client_credentials" },
+                    { "client_id", clientId },
+                    { "client_secret", clientSecret },
+                    { "scopes", string.Join(" ", scopes) }
+                };
+
+                request.SetRequestLine(HttpPipelineMethod.Put, authUri);
+
+                ReadOnlyMemory<byte> content = Serialize(body);
```

Comment:
Creating a hash-table just to write it to json writer is a bit wasteful. Why not simply pass these parameters directly to the Serialize method? Similarly on deserialize